src/general/smoothing_kernels.jl

```diff
@@ -3,7 +3,8 @@ abstract type SmoothingKernel{NDIMS} end
 @inline Base.ndims(::SmoothingKernel{NDIMS}) where {NDIMS} = NDIMS
 
 @inline function kernel_grad(kernel, pos_diff, distance, h)
-    distance < sqrt(eps()) && return zero(pos_diff)
+    # Use `eps` relative to `h` and with the type of `h` to allow for `Float32`
+    distance < sqrt(eps(h)) && return zero(pos_diff)
```

Comment:
Shouldn't this be also typeof(h)?

Reply:
See the comment. This is using `eps` relative to `h`. So when scaling a problem down by a factor of a million, this will also scale down the `eps` accordingly.

Reply:
But is this what we want? This will result in different cutoffs with variable smoothing_length.
```
julia> eps(Float32)
1.1920929f-7

julia> sqrt(eps(Float32(0.1)))
8.6316744f-5

sqrt(eps(Float32(100)))
0.0027621358f0
```

Reply:
If you have particle refinement in mind, then we restrict the variable smoothing length to 5...20% in the neighborhood. I think scaling `eps` is reasonable then.

Reply:
> But is this what we want? This will result in different cutoffs with variable smoothing_length.

Well, this is eps. So the rounding error for `distance2` will be in the order of `eps(smoothing_length)`. When two larger particles interact, the rounding error will be larger because everything is scaled, so it's natural that we need a larger tolerance.